arviz/plots/backends/matplotlib/distplot.py

```diff
@@ -91,8 +91,11 @@ def _histplot_mpl_op(values, values2, rotated, ax, hist_kwargs):
         raise NotImplementedError("Insert hexbin plot here")
 
     bins = hist_kwargs.pop("bins")
+    if type(values).__name__ == "DataArray":
+        ax.hist(values.values.flatten(), bins=bins, **hist_kwargs)
+    else:
+        ax.hist(values, bins=bins, **hist_kwargs)
```

Comment:
Here I'd use only `values = np.asarray(values).faltten()` (and probably the same in bokeh file, haven't checked though). This will accept DataArrays stright away.

Moreover, it is probably a good idea to add some errors in the general file `plots/distplot.py` in a similar way to [plot_kde](https://github.com/arviz-devs/arviz/blob/master/arviz/plots/kdeplot.py#L164-L170), so users get some informative error if they try to use the function on an inference data or a dataset.

Reply:
Yes, this solution is also working fine. 

For, the second suggestion, Are following error appropriate?

`          "Cannot directly convert xarray.Dataset to numpy array. Instead, create an xarray.DataArray first or Use plot_posterior, plot_density, plot_joint"
            "or plot_pair instead of plot_dist"
      `
and
`  
    " Inference Data object detected. Use plot_posterior instead of plot_dist"`